cli/util/upload_bin.go

```diff
@@ -151,24 +150,14 @@ func uploadBin(ctx *cli.Context) error {
 		return cli.Exit(fmt.Sprintf("failed to get current block height from RPC: %v", err), 1)
 	}
 	fmt.Fprintln(ctx.App.Writer, "Chain block height:", currentBlockHeight)
-
-	oldestMissingBlockIndex, errBlock := fetchLatestMissingBlockIndex(ctx.Context, p, containerID, acc.PrivateKey(), attr, int(currentBlockHeight), maxParallelSearches, maxRetries)
-	if errBlock != nil {
-		return cli.Exit(fmt.Errorf("failed to fetch the oldest missing block index from container: %w", errBlock), 1)
-	}
-	fmt.Fprintln(ctx.App.Writer, "First block of latest incomplete batch uploaded to NeoFS container:", oldestMissingBlockIndex)
-
-	if !ctx.Bool("skip-blocks-uploading") {
-		err = uploadBlocks(ctx, pWrapper, rpc, signer, containerID, acc, attr, oldestMissingBlockIndex, uint(currentBlockHeight), homomorphicHashingDisabled, numWorkers, maxRetries, debug)
-		if err != nil {
-			return cli.Exit(fmt.Errorf("failed to upload blocks: %w", err), 1)
-		}
-		oldestMissingBlockIndex = int(currentBlockHeight) + 1
+	index, partUploadedBlocks, err := bufferedIndexFile(ctx, pWrapper, containerID, acc, signer, indexFileSize, attr, indexAttrKey, maxParallelSearches, maxRetries)
+	if err != nil {
+		return cli.Exit(fmt.Errorf("failed to upload index files: %w", err), 1)
 	}
 
-	err = uploadIndexFiles(ctx, pWrapper, containerID, acc, signer, uint(oldestMissingBlockIndex), attr, homomorphicHashingDisabled, maxParallelSearches, maxRetries, debug)
+	err = uploadBlocksAndIndexFiles(ctx, pWrapper, rpc, signer, containerID, acc, attr, indexAttrKey, partUploadedBlocks, int(index*indexFileSize), indexFileSize, uint(currentBlockHeight), homomorphicHashingDisabled, numWorkers, maxRetries, debug)
```

Comment:
Adjust `int(index*indexFileSize)` code, pass only `i`, adjust the meaning of `uploadedHeight int` function parameter and its naming.